finality-aleph/src/network/service.rs

```diff
@@ -843,6 +896,312 @@ mod tests {
+            });
+
+        // We do this only to make sure that NotificationStreamOpened events are handled
+        test_data.wait_for_events_handled().await;
+
+        test_data
+            .mock_io
+            .messages_for_user
+            .unbounded_send((
+                message_1.clone(),
+                DataCommand::SendTo(identity.1, Protocol::Generic),
+            ))
+            .unwrap();
+
+        test_data
+            .mock_io
+            .messages_for_user
+            .unbounded_send((
+                message_2.clone(),
+                DataCommand::SendTo(identity.1, Protocol::Generic),
+            ))
+            .unwrap();
+
+        let expected = (message_2.encode(), identity.1, Protocol::Generic.name());
+
+        assert_eq!(
+            test_data
+                .network
+                .send_message
+                .next()
+                .await
+                .expect("Should receive message"),
+            expected,
+        );
+
+        test_data.cleanup().await
+    }
+
+    #[tokio::test]
+    async fn test_generic_create_sender_error_many_peers() {
+        let mut test_data = TestData::prepare().await;
+
+        let all_authorities_n = 4;
+        let closed_authorities_n = 2;
+        for _ in 0..closed_authorities_n {
+            test_data
+                .network
+                .create_sender_errors
+                .lock()
+                .push_back(MockSenderError::SomeError);
+        }
+
+        let identities: Vec<_> = (0..4)
+            .map(|_| MockNetworkIdentity::new().identity())
+            .collect();
+        let message: Vec<u8> = vec![1, 2, 3];
+
+        identities.iter().for_each(|identity| {
+            test_data
+                .network
+                .emit_event(Event::NotificationStreamOpened {
+                    protocol: Cow::Borrowed(ALEPH_PROTOCOL_NAME),
+                    remote: identity.1.into(),
+                    negotiated_fallback: None,
+                    role: ObservedRole::Authority,
+                })
+        });
+
+        // We do this only to make sure that NotificationStreamOpened events are handled
+        test_data.wait_for_events_handled().await;
+
+        identities.iter().for_each(|identity| {
+            test_data
+                .mock_io
+                .messages_for_user
+                .unbounded_send((
+                    message.clone(),
+                    DataCommand::SendTo(identity.1, Protocol::Generic),
+                ))
+                .unwrap();
+        });
+
+        let broadcasted_messages = HashSet::<_>::from_iter(
+            test_data
+                .network
+                .send_message
+                .1
+                .lock()
+                .await
+                .by_ref()
+                .take(all_authorities_n - closed_authorities_n)
+                .collect::<Vec<_>>()
+                .await
+                .iter()
+                .cloned(),
```

Comment:
Shouldn't the last two lines here be `into_iter` instead?

Reply:
I don't know, you accepted it in other pr xd

Reply:
Done